### PR TITLE
SEO improvements

### PR DESCRIPTION
**Pull request by vnegrisolo**
# Motivation

we found so much trash indexed urls in the google search console that we decided to review SEO for `tilex` and we come up with this PR. We are expecting to reduce the number of indexed page from 3.5K pages to about 600. That might improve the rank of our posts on search engines in general.

## Changes

- [x] add the sitemap entry to the `robots.txt` file
- [x] add channels show page to the `sitemap.xml`
- [x] remove some pages from bots index via `robots.txt`
- [x] set a few meta robots `noindex`
- [x] set canonical url on all pages and ignore query string by default
- [x] refactor: avoid unecessary join on channels random
- [x] refactor: reuse `Tilex.Pageable` on all controllers

---

# I'll revisit Google Search Console in a few weeks to check the new index situation

| before | after |
| --- | --- |
| image | |
| image | |
| image | |

## See also

https://search.google.com/

### CODE REVIEW

**Review by tmock12 — APPROVED** [review]

**Comment by vnegrisolo** [comment]
thanks @tmock12 I am still working on this PR though. I want to add some noindexes and check it out canonicals etc

**Comment by vnegrisolo** [comment]
@tmock12 now this PR is ready for review

File: lib/tilex/posts.ex [hunk 1]
```diff
@@ -35,15 +35,14 @@ defmodule Tilex.Posts do
...
+    channel_id = channel.id
     posts_count = 1
 
     query =
       from(
         post in Post,
         order_by: fragment("random()"),
-        join: channel in assoc(post, :channel),
-        where: channel.name == ^channel_name,
+        where: post.channel_id == ^channel_id,
```
Comment (tmock12) [inline lib/tilex/posts.ex:45]: Given it's `limit: 1`, would it be better to leave the join, remove the separate Channel DB `get_by!/2` call and just return the single posts channel? Then you'd only have a single DB call. ie something like::

```elixir
 def random_post_by_channel(channel_name) do
    posts_count = 1

    query =
      from(
        post in Post,
        order_by: fragment("random()"),
        join: channel in assoc(post, :channel),
        where: channel.name == ^channel_name,
        limit: ^posts_count,
        preload: [:channel, :developer]
      )
    post = Repo.one(query)

    {[post], posts_count, post.channel}
  end
```

Comment (vnegrisolo) [reply lib/tilex/posts.ex]: that's even better, thanks @tmock12 !

**Review by avogel3 — APPROVED** [review]

**Pull request merged**